docs/docs/04-features.md

```diff
@@ -125,56 +123,36 @@ Note that this only works for directories that have been mounted via a `mount:*`
 
 ### JSX
 
-#### Compile to JavaScript
-
+Snowpack has built-in support to handle `.jsx` & `.tsx` source files in your application. 
 
-Snowpack automatically builds all `.jsx` & `.tsx` files to JavaScript during development and production builds. 
+**Note: Snowpack's default build does not support JSX in  `.js`/`.ts` files.** If you can't use the `.jsx`/`.tsx` file extension, you can create a Babel config file and use the official Babel plugin to build your application:
 
-**Note: Snowpack's default build supports JSX with both React & Preact as long as a React/Preact import exists somewhere in the file.** To set a custom JSX pragma, you can configure our default esbuild yourself:
 
 ```js
 // snowpack.config.json
-// Optional: Define your own JSX factory/fragment
+// [npm install --save-dev @snowpack/plugin-babel]
 {
-  "scripts": {
-    "build:tsx": "esbuild --jsx-factory=h --jsx-fragment=Fragment --loader=tsx"
-  }
-}
-```
-
-**Note: Snowpack's default build does not support JSX in  `.js`/`.ts` files.** You'll need to define your own build script to support this. You can define your own JSX->JavaScript build step via a [Build Script integration](#build-scripts).
-
-```js
-// snowpack.config.json
-// Optional: You can define your own JSX build step if you'd like.
-{
-  "scripts": {
-    "build:jsx": "babel --filename $FILE",
-  }
+  "plugins": ["@snowpack/plugin-babel"]
 }
 ```
 
 ### TypeScript
 
 #### Compile to JavaScript
 
-Snowpack automatically builds all `.ts` & `.tsx` files to JavaScript. Snowpack will not perform any type checking by default (see below), only building from TS->JS.
-
-You could also choose to define your own TSX->JavaScript build step via a [Build Script integration](#build-scripts).
+Snowpack has built-in support to handle `.ts` & `.tsx` source files in your application. If you prefer to use Babel, you can instead create a Babel config file and use the official Babel plugin to build your application:
 
 ```js
 // snowpack.config.json
-// Optional: You can define your own TS build step if you'd like.
+// [npm install --save-dev @snowpack/plugin-babel]
 {
-  "scripts": {
-    "build:ts,tsx": "babel --filename $FILE",
-  }
+  "plugins": ["@snowpack/plugin-babel"]
 }
 ```
 
-#### Type Checking During Development
+#### Type-Checking During Development
```

Comment:
I think only the adjective needs a hyphen, not the noun form, e.g. `I am type checking a type-checked application.`